dask_cuda/tests/test_dask_cuda_worker.py

```diff
@@ -58,12 +57,43 @@ def get_visible_devices():
                 for v in result.values():
                     del expected[v]
 
+                assert len(expected) == 0
+
+
+def test_memory_limit_and_nthreads(loop):  # noqa: F811
```

Comment:
This test had to be split from `test_cuda_visible_devices` because `data` is now determined _per-device_ and not globally based on device 0. Therefore, testing devices that are unavailable becomes impossible as NVML is unable to query device memory from devices that don't exist, and `DASK_CUDA_TEST_DISABLE_DEVICE_SPECIFIC` was specifically introduced to create a sort of a mock test.